File: src/java/org/orbeon/oxf/resources/handler/HTTPHandler.java
/**
 *  Copyright (C) 2006 Orbeon, Inc.
 *
 *  This program is free software; you can redistribute it and/or modify it under the terms of the
 *  GNU Lesser General Public License as published by the Free Software Foundation; either version
 *  2.1 of the License, or (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *  See the GNU Lesser General Public License for more details.
 *
 *  The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
 */
package org.orbeon.oxf.resources.handler;

import java.io.IOException;
import java.net.URL;
import java.net.URLConnection;
import java.net.URLStreamHandler;

/**
 * Implements a URLStreamHandler for HTTP to be passed to the URL constructor.
 * The Apache HTTP Client library will be used to handle the URL.
 */
public class HTTPHandler extends URLStreamHandler  {

    protected URLConnection openConnection(URL url) throws IOException {
        return new HTTPURLConnection(url);
    }
}

File: src/main/scala/org/orbeon/oxf/resources/handler/HTTPHandler.scala
/**
 * Copyright (C) 2006 Orbeon, Inc.
 *
 * This program is free software; you can redistribute it and/or modify it under the terms of the
 * GNU Lesser General Public License as published by the Free Software Foundation; either version
 * 2.1 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
 */
package org.orbeon.oxf.resources.handler

import java.net.{URL, URLStreamHandler}

import org.orbeon.oxf.http.{ApacheHttpUrlConnection, PropertiesApacheHttpClient}

class HTTPHandler extends URLStreamHandler {
    protected def openConnection(url: URL) =
        new ApacheHttpUrlConnection(url)(PropertiesApacheHttpClient)
}